### DISCUSSION TITLE
Dragging objects with DragController (Three.js + @enable3d/ammo-physics)

### DISCUSSION THREAD

**Discussion by GRicciardi00** [enable3d/enable3d#379] [Help / Questions]
Hi, me and my colleagues are working Tower of Hanoi game, I'm the one who's doing the physics system.
After several days of working to implement the physics directly with Ammo with bad results, I discovered this fantastic library.
Currently I set up the physics bodies, the problem is that I when I try to drag the objects I can't move them.
Does anyone have a solution for this?
> import * as THREE from 'three';
import { OrbitControls } from "three/addons/controls/OrbitControls.js";
import Scene from './class/scene.js';
import * as ENABLE3D from '@enable3d/ammo-physics';
let scene;
let selectedObject; 
let makingMove = false;
let movesMade = 0; 
let Invalid = false;
let raycaster, CURRENTCOLOR;
const clock = new THREE.Clock()
let physics
const { AmmoPhysics, PhysicsLoader } = ENABLE3D
const pointer = new THREE.Vector2();
const ThreeScene = () => {
    //INIT
    scene = new Scene();
    raycaster = new THREE.Raycaster();
    raycaster.params.Line.threshold = 3;
    //setup main orbit control
    const orbitControls = new OrbitControls( scene.camera, scene.renderer.domElement );
    scene.setUpControl();
    CURRENTCOLOR = scene.disks_mashes[0].material.color.getHex();
    physics = new AmmoPhysics(scene.scene);
    //Add physics to the scene
    physics.debug.enable();
    // static ground
    physics.add.existing(scene.table, {mass: 0});
    //Add physics to the disks
    for (let i = 0; i < scene.disks.length; i++) {
        physics.add.existing(scene.disks_mashes[i], {mass: 1});
    }
    scene.controls.addEventListener( 'drag', function(event){
        selectedObject = event.object
        render();
    } );
    document.addEventListener( 'pointermove', onPointerMove );
    scene.controls.addEventListener('dragstart', function (event) {
        //disable orbit control
        makingMove = true;
        orbitControls.enabled = false;
        selectedObject = event.object;
        const intersects_0 = raycaster.intersectObjects( scene.disks_mashes, true );
        if ( intersects_0.length > 0 ) {
            intersects_0[0].object.material.color.set( 0xff0000 );
        }
    });
    scene.controls.addEventListener('dragend', function (event) {
        //enable orbit control
        makingMove = false;
        movesMade +=1;
        orbitControls.enabled = true;
        const selectedObject = event.object;
        selectedObject.material.color.setHex( CURRENTCOLOR );
    });
    window.addEventListener( 'resize', onWindowResize );
    animate();
}
function onWindowResize() {
    scene.camera.aspect = window.innerWidth / window.innerHeight;
    scene.camera.updateProjectionMatrix();
    scene.renderer.setSize( window.innerWidth, window.innerHeight );
    render();
}
//Raycaster logic
function onPointerMove( event ) {
    pointer.x = ( event.clientX / window.innerWidth ) * 2 - 1;
	pointer.y = - ( event.clientY / window.innerHeight ) * 2 + 1;
    raycaster.setFromCamera( pointer, scene.camera );
}
function render(event) {
    physics.update(clock.getDelta() * 1000)
    physics.updateDebugger()
    scene.renderer.render( scene.scene, scene.camera );
}

This is a screenshot of what we have done so far, the red disk is the one selected
<img width="652" alt="Screenshot 2024-01-17 alle 14 44 58" src="https://github.com/enable3d/enable3d/assets/102236495/84b636c8-ea1b-4fe2-bb31-540cb1044fad">

**Comment by scenaristeur** [comment]
Hi i don't find in your code how disk are created, but did you read the doc ... ??? This may help https://enable3d.io/docs.html#physics-body
